flang/lib/Semantics/resolve-directives.cpp

```diff
@@ -409,6 +409,19 @@ class OmpAttributeVisitor : DirectiveAttributeVisitor<llvm::omp::Directive> {
   }
   void Post(const parser::OpenMPDepobjConstruct &) { PopContext(); }
 
+  bool Pre(const parser::OpenMPFlushConstruct &x) {
+    PushContext(x.source, llvm::omp::Directive::OMPD_flush);
+    for (auto &arg : x.v.Arguments().v) {
+      if (auto *locator{std::get_if<parser::OmpLocator>(&arg.u)}) {
+        if (auto *object{std::get_if<parser::OmpObject>(&locator->u)}) {
+          ResolveOmpObject(*object, Symbol::Flag::OmpDependObject);
```

Comment:
Assuming this is a copy paste error. This is not a `DependObject`, so I think adding the `Symbol::Flag::OmpDependObject` is probably not right.

What do we want to do here? Is it just resolving the names in the flush argument list to the appropriate Fortran declarations? If so, we should use `ResolveName`.

Reply:
If common blocks are not handled in `ResolveName` then you can probably use `OmpFlushed` as the Symbol in `ResolveOmpObject`.

Reply:
Yes it was a copy and paste error. Thanks for catching this.

I had assumed that given an object, the right way to resolve it was to call ResolveOmpObject. I have now changed to include only the relevant parts of the code inline.